dotcom-rendering/src/lib/useBraze.ts

```diff
@@ -24,8 +25,11 @@ export const useBraze = (
 	brazeMessages: BrazeMessagesInterface | undefined;
 	brazeCards: BrazeCardsInterface | undefined;
 } => {
-	const { data, error } = useSWRImmutable('braze-message', () =>
-		buildBrazeMessaging(idApiUrl),
+	const signedInStatus = useIsSignedInStatus();
+
+	const { data, error } = useSWRImmutable(
+		signedInStatus !== 'Pending' ? 'braze-message' : null,
+		() => buildBrazeMessaging(idApiUrl, signedInStatus === 'SignedIn'),
```

Comment:
Just to make sure I understand the difference here - is it correct that `buildBrazeMessaging` will generally be called twice now per page view now - 1. once while we're waiting for the signed-in status i.e. when it's pending and the cache key passed to `useSWRImmutable` will be `null`, and then 2. again when the signed in status is known where the cache key will be `braze-message`? If that's the case I don't _think_ it'll be an issue, but it's a change from what's in prod currently where the guarantee is that we'll only ever call `buildBrazeMessaging` once. I'd be keen to check this in the CODE environment to make sure everything's behaving as expected.

Hmm actually thinking a little more about this, in the first call (when we don't yet know the signed in status) `buildBrazeMessaging` will return null objects like `NullBrazeMessaging` which behave like the real thing but will never return messages. I think in places where we call `useBraze` we expect that when a defined value is returned that's the final answer. See usage of `useOnce` with `brazeMessages` as a dependency in `SlotBodyEnd` for example. So when we get back the null objects, the `useOnce` callback will be invoked but no messages will be shown. If the user is signed in/consented etc then `useBraze` will subsequently return the real BrazeMessages instance, but by that point our `useOnce` has already run so nothing will happen.

Maybe another option is to call `useSWRImmutable` (and therefore also `buildBrazeMessaging`) conditionally, only when the signed in status is known? And until that point return undefined values for `brazeMessages` and `brazeCards`?

Reply:
> My understanding is that if you pass null as the key to an SWR hook, then SWR won't start the request specified by the fetcher function: https://swr.vercel.app/docs/conditional-fetching

Ah great, I did try (and fail!) to find documentation for the case where you pass `null` as the key so that's good to know. In that case yep I think your description here sounds right 👍

> useSWRImmutable is a hook and I don't think it's possible to call hooks conditionally?

Ah of course yep, don't know what I was thinking when I wrote that! I guess if we _did_ need to do something similar the condition could be in the `useSWRImmutable` callback (I think the key is passed to the callback, so in the case it's `null`, return undefined/do nothing). But I think you're right, given the behaviour you described it doesn't sounds like this will be needed.

Reply:
Just for reference, we've been able to confirm that `buildBrazeMessaging` runs only once as intended.